ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureActivity.java

```diff
@@ -36,11 +38,29 @@ public void onValidated(Context context, ValidateResponse validateResponse, Stri
                 handleValidated(validateResponse, s);
             }
         });
-        onCreateInternal(cardinalClient);
+
+        new Handler(Looper.getMainLooper()).post(new Runnable() {
```

Comment:
This works, but I'm wondering if there's a better way? We basically need to defer the call to `launchCardinalAuthChallenge()` so that `handleValidated()` gets a chance to run after a process kill.